synthesizer/src/store/helpers/rocksdb/internal/mod.rs

```diff
@@ -129,50 +129,67 @@ impl Database for RocksDB {
     }
 }
 
+#[cfg(feature = "testing")]
 impl RocksDB {
     /// Opens the test database.
-    #[cfg(test)]
-    fn open_testing(temp_dir: std::path::PathBuf, dev: Option<u16>) -> Result<Self> {
-        let database = {
+    fn open_testing(path: Option<std::path::PathBuf>) -> Result<Self> {
+        use once_cell::sync::Lazy;
+        use std::collections::HashMap;
+
+        // A global mapping of test databases.
+        static DATABASES: Lazy<Mutex<HashMap<std::path::PathBuf, RocksDB>>> = Lazy::new(|| Mutex::new(HashMap::new()));
```

Comment:
do we need to support multiple databases? so far we've only ever used multiple maps, but all belonging to a single database